### ISSUE TITLE
Would you consider abandoning the constructor function introduced in sequelize-connect@2?

### ISSUE THREAD

**Issue by bookercodes** [jspizziri/sequelize-connect#12]
I _much_ prefer the `sequelize-connect@1` API. _Would you consider abandoning the constructor function introduced in `sequelize-connect@2`?_

I have a number of arguments _against_ the constructor pattern in this context:

**Argument 1**

It is more convenient and idiomatic to write:

``` js
import db from 'sequelize-connect'
```

... than it is to write:

``` javascript
import orm from 'sequelize-connect'
const db = new Connect()
```

**Argument 2**

[You don't need to _manually_ implement the singleton pattern in Node](http://stackoverflow.com/questions/13179109/singleton-pattern-in-nodejs-is-it-needed). I think doing so adds unnecessary complexity and is _not_ idiomatic. What you did in `sequelize-connext@1` was _so_ much nicer to read and use.

**Argument 3**
#9 :wink:

**Argument 4**

I think a constructor should only _construct_ the objects state based on the given `arguments`. It should _not_ perform an operation like connecting to the database. Take [`sequelize`](https://www.npmjs.com/package/sequelize) as an example. When you write `new Sequelize(...)`, you're not connecting to the database. 

**Argument 5**

In my opinion, the API is _very_ unnatural. 

Consider this code:

``` javascript
const orm = new Connection('schema', 'root', 'some password', { }, discover, matcher)
```

I think this code is natural enough but 

``` javascript
const orm = new Connection()
```

... is just really _weird_. 

It's unnatural and the use of `new` _implies_ that a new connection is being created. When I see the singleton pattern implemented, I usually see it done with [an `instance` property](http://csharpindepth.com/Articles/General/Singleton.aspx) (granted, that is C# where you can't change what a constructor returns, but the first argument still applies)

**Argument 6**

The constructor takes a lot of arguments. It's hard to remember the order and skipping arguments isn't intuitive. _It's a prime candidate for a [parameter object](http://refactoring.com/catalog/introduceParameterObject.html)._ Thing is, `sequelize-connect@1` didn't suffer from this problem -- use the old pattern and this problem goes away :wink:

**Argument 7**

In a code base of mine using `sequelize-connect@1` I wrote the following:

``` javascript
import db from 'sequelize-connect'

beforeEach(function connectToAndResetTestDatabase () {
  db.logger.level = null // Disable sequelize-connect logging
  db.discover = path.join(__dirname, '../../../source/models')
  db.matcher = function shouldImportModel (modelFileName) {
    return true
  }
  return db.connect(config.database, config.username, config.password, {
    force: true,
    logging: false
  }) 
})
it('badly named test', async () => {
  config.screencastsPerPage = 2
  const screencasts = require('./screencatsFixture.json')
  for (const screencast of screencasts) {
    await db.models.screencast.createScreencast(screencast)
  }
  // ...
})

afterEach(function disconnectFromTestDatabase () {
  db.sequelize.close()
})
```

To express the same thing with `sequelize-connect@2` is less eloquent:

``` javascript
import Connection from 'sequelize-connect'
let db

beforeEach(function connectToAndResetTestDatabase () {
  const discover = path.join(__dirname, '../../../source/models')
  const matcher = function shouldImportModel (modelFileName) {
    return true
  }
  return db.connect(config.database, config.username, config.password, {
    force: true,
    logging: false
  }).then(function(instance) {
   db = instance // I think this is the equivalent?
  })
})
it('badly named test', async () => {
  config.screencastsPerPage = 2
  const screencasts = require('./screencatsFixture.json')
  for (const screencast of screencasts) {
    await db.models.screencast.createScreencast(screencast)
  }
  // ...
})

afterEach(function disconnectFromTestDatabase () {
  db.sequelize.close()
})
```

**Comment by bookercodes** [comment]
If our visions do not align then that is _totally_ cool :sunglasses: 

I can always maintain a fork or create a similar lib but I would much rather work together

**Comment by eredisg** [comment]
+1, @1.0 was a lot better in my opinion.

**Comment by jspizziri** [comment]
@CodeBeasty,

You can always install the `1.0` version in your project, rather than `2.0`. Additionally, I've been very busy so I haven't had any time to work on an improved version of `2.0`. If either you or @alexbooker wanted to submit a PR for an improved version of `2.0` (or perhaps it would technically be a `3.0`, I'm all ears.

Thanks!

**Comment by jspizziri** [comment]
@bookercodes,

Closing this for now. If I see a PR in the future perhaps everyone can be happy :) 


### For anyone else interested in `1.0` interface 

I recommend simply installing the `1.0` version of the package thusly:

```
npm install sequelize-connect@^1.0 --save
```

Voila! No need to maintain a separate branch. If any issues are fixed in `2.0` they should be backported to `1.0`
